### PR TITLE
AB testing: implement the first A/B experiment

### PR DESCRIPTION
**Pull request by leandroalonso**
Related PR: https://github.com/Automattic/Automattic-Tracks-iOS/pull/174

After our successful A/A test (https://github.com/wordpress-mobile/WordPress-iOS/pull/15789) it's time for our first A/B test!

This PR implements the first A/B test within the FAB button (under My Site).

Also, it fixes a bug identified in the previous version (which was requesting the `assignments` endpoint even without any active tests)

| Control | Treatment |
| ------- | ----------- |
| Simulator Screen Shot - iPhone 11 - 2021-03-18 at 16 22 06 | Simulator Screen Shot - iPhone 11 - 2021-03-18 at 16 23 23 |

### To test

1. Open the app
2. Tap the FAB, you should see the following order: "Blog post", "Site page" and "Story post"
3. Close the app

Now you'll need to use a tool like Charles to intercept the proxy.

1. Open the app again
2. Intercept the assignments request and change `wpios_create_menu_story_first` to `treatment`
3. 2. Tap the FAB
4. You should see the following order: "Story post", "Blog post" and "Site page"

You can see this test in the following video:

https://user-images.githubusercontent.com/7040243/111689322-a5582580-880a-11eb-9cda-429392162420.mov

### Exposure event

The PR is still missing the exposure event, I'll add it before merging it.

### CODE REVIEW

**Comment by leandroalonso** [comment]
@aaronyan I hope that by looking at the video and the images you would be able to validate this PR. Also, let me know once we define the exposure event so I can update the code with it. :)

File: WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonActionSheet.swift [hunk 1]
```diff
@@ -11,6 +11,15 @@ class CreateButtonActionSheet: ActionSheetViewController {
     }
 
     init(actions: [ActionSheetItem]) {
+
+        /// A/B test: display story first
+        var actions = actions
+        if !actions.filter({ $0 is StoryAction }).isEmpty
+            && ABTest.storyFirst.variation == .treatment(nil) {
+            let storyAction = actions.remove(at: 2)
+            actions.insert(storyAction, at: 0)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonActionSheet.swift]: TIL there's a swapAt(_:_:) method
https://developer.apple.com/documentation/swift/array/2893281-swapat

Totally up to you if you want to do remove/insert or swapAt 😀

**Review by momo-ozawa — APPROVED** [review]
LGTM!

File: WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonActionSheet.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonActionSheet.swift]: TIL too! Thanks! Changed in 3502df59f19cc6e6629baf5f2ddf177c82720f0c

**Pull request merged**